libs/datamodel/connectors/dml/src/model.rs

```diff
@@ -323,6 +299,44 @@ impl Model {
 
         has_field(self, "createdAt") && has_field(self, "updatedAt")
     }
+
+    pub fn field_is_unique(&self, name: &str) -> bool {
+        self.indices
+            .iter()
+            .any(|i| i.is_unique() && i.fields == vec![name.to_string()])
+    }
+
+    pub fn field_is_unique_and_defined_on_field(&self, name: &str) -> bool {
+        self.indices
+            .iter()
+            .any(|i| i.is_unique() && i.fields == vec![name.to_string()] && i.defined_on_field)
+    }
+
+    pub fn field_is_primary(&self, field_name: &str) -> bool {
+        matches!(&self.primary_key, Some(PrimaryKeyDefinition{ fields, .. }) if fields == &vec![field_name.to_string()])
+    }
+    pub fn field_is_primary_and_defined_on_field(&self, field_name: &str) -> bool {
+        matches!(&self.primary_key, Some(PrimaryKeyDefinition{ fields, defined_on_field , ..}) if fields == &vec![field_name.to_string()] && *defined_on_field)
+    }
+
+    pub fn field_is_auto_generated_int_id(&self, name: &str) -> bool {
+        let field = self.find_scalar_field(name).unwrap();
+
+        let is_autogenerated_id =
+            matches!(field.default_value, Some(DefaultValue::Expression(_)) if self.field_is_primary(name));
+
+        let is_an_int = matches!(field.field_type, FieldType::Scalar(ScalarType::Int, _, _));
+
+        is_autogenerated_id && is_an_int
+    }
+
+    pub fn id_field_names(&self) -> Vec<String> {
```

Comment:
In general, cloning a vector out from a method is not very nice. In most cases we just want to iterate over the field names, and this could be either `&[&str]` or `impl Iterator<Item = &str>`...

Reply:
Currently there is only one user of the function and there we need a Vec. And that one the QE guys can hopefully remove to fully get rid of this.